pkg/controller/volume/attachdetach/populator/desired_state_of_world_populator.go

```diff
@@ -113,6 +113,21 @@ func (dswp *desiredStateOfWorldPopulator) populatorLoopFunc() func() {
 // Iterate through all pods in desired state of world, and remove if they no
 // longer exist in the informer
 func (dswp *desiredStateOfWorldPopulator) findAndRemoveDeletedPods() {
+
+	// check if the existing volumes changes its attachability
+	for _, volumeToAttach := range dswp.desiredStateOfWorld.GetVolumesToAttach() {
```

Comment:
Would it make sense to move this to the end (first remove deleted pods, then check for this)?